### PR TITLE
Reporting: Support job specific data

### PR DESCRIPTION
**Pull request by psiinon**
Part of https://github.com/zaproxy/zaproxy/issues/6483

This adds a new Passing Rules section to the HTML plus report:

Screenshot_2021-04-06 ZAP Scanning Report

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
Passing as in they were run but didn't generate an alert?

**Comment by psiinon** [comment]
Yes.
For passive scan rules that will be all of the enabled rules that have not reported an alert (that is in the scope of the report).
For active scan rules it will be all of the rules that were run in the last active scan.

File: addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/job-ascan.html [hunk 1]
```diff
@@ -37,6 +39,14 @@ <H1>Automation Framework - activeScan Job</H1>
         threshold:                     # String: The Alert Threshold for this rule, one of Off, Low, Medium, High, default: Medium
 </pre>
 
+<H2>Job Data</H2>
+The following class will be made available to add-ons that provide access to the Job Data such as the Reporting add-on.
+Note that in this case the data is for the last Active Scan, regardless of whether it was started by the Automation Framework, the UI or the API.
```
Comment (kingthorin) [inline addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/job-ascan.html:44]:
```suggestion
Note that in this case the data is for the last Active Scan, regardless of whether it was started by the Automation Framework, the UI, or the API.
```

File: addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/job-pscanwait.html [hunk 2]
```diff
@@ -13,12 +13,22 @@ <H1>Automation Framework - passiveScan-wait Job</H1>
...
+
 <pre>
   - type: passiveScan-wait             # Passive scan wait for the passive scanner to finish
     parameters:
       maxDuration: 5                   # Int: The max time to wait for the passive scanner, default: 0 unlimited
 </pre>
 
+<H2>Job Data</H2>
+The following class will be made available to add-ons that provide access to the Job Data such as the Reporting add-on.
+Note that in this case the data is for the currently enabled Passive Scan rules, regardless of whether they have been used as a result of the Automation Framework, the UI or the API.
```
Comment (kingthorin) [inline addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/job-pscanwait.html:26]:
```suggestion
Note that in this case the data is for the currently enabled Passive Scan rules, regardless of whether they have been used as a result of the Automation Framework, the UI, or the API.
```

**Comment by psiinon** [comment]
Oops. Can we have a formatting rule which checks for missing Oxford comms? :P

**Review by kingthorin — APPROVED** [review]

File: addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/job-ascan.html [hunk 3]
```diff
@@ -37,6 +39,14 @@ <H1>Automation Framework - activeScan Job</H1>
         threshold:                     # String: The Alert Threshold for this rule, one of Off, Low, Medium, High, default: Medium
 </pre>
 
+<H2>Job Data</H2>
+The following class will be made available to add-ons that provide access to the Job Data such as the Reporting add-on.
+Note that in this case the data is for the last Active Scan, regardless of whether it was started by the Automation Framework, the UI, or the API.
```
Comment (thc202) [inline addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/job-ascan.html:44]: `for` → `from`?

File: addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/job-pscanwait.html [hunk 4]
```diff
@@ -13,12 +13,22 @@ <H1>Automation Framework - passiveScan-wait Job</H1>
...
+
 <pre>
   - type: passiveScan-wait             # Passive scan wait for the passive scanner to finish
     parameters:
       maxDuration: 5                   # Int: The max time to wait for the passive scanner, default: 0 unlimited
 </pre>
 
+<H2>Job Data</H2>
+The following class will be made available to add-ons that provide access to the Job Data such as the Reporting add-on.
+Note that in this case the data is for the currently enabled Passive Scan rules, regardless of whether they have been used as a result of the Automation Framework, the UI, or the API.
```
Comment (thc202) [inline addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/job-pscanwait.html:26]: Same comment.

File: addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/create.html [hunk 5]
```diff
@@ -127,9 +127,9 @@ <H3>reportData</H3>
 	class that defines the data to be included in the report. This can be
 	used, for example, to tell if there were no level alerts of a specific
 	level or if they were deliberately excluded.
-	<p>The plan is to allow any add-ons to make additional data items
-		available. This help page will be updated when that feature is
-		available.</p>
+	<p>Automation jobs can now make additional data available to the
```
Comment (thc202) [inline addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/create.html:130]: Remove now? It does not really matter "when" going forward.

File: addOns/reports/reports.gradle.kts [hunk 6]
```diff
@@ -15,6 +15,18 @@ zapAddOn {
         author.set("ZAP Dev Team")
         url.set("https://www.zaproxy.org/docs/desktop/addons/report-generation/")
         extensions {
+            register("org.zaproxy.addon.reports.ExtensionReports") {
+                classnames {
+                    restricted.set(listOf("org.zaproxy.addon.reports.automation"))
+                }
+                dependencies {
+                    addOns {
+                        register("automation") {
```
Comment (thc202) [inline addOns/reports/reports.gradle.kts:24]: This means that to create reports you need the automation add-on.

Comment (psiinon) [reply addOns/reports/reports.gradle.kts:24]: Thats correct. The plan is to include both as standard for the next release.

**Comment by psiinon** [comment]
Hopefully fixed those comments.

File: addOns/reports/reports.gradle.kts (hunk 6 above)
Comment (thc202) [reply addOns/reports/reports.gradle.kts:24]: Is this temporary? Ideally it should not need to depend.

**Review by thc202 — APPROVED** [review]

**Pull request merged**